colossalai/booster/plugin/torch_fsdp_plugin.py

```diff
@@ -0,0 +1,224 @@
+        """
+        Save model to checkpoint but only on master process.
+        """
+        # the model should be unwrapped in self.load_model via ModelWrapper.unwrap
+
+        if version.parse(torch.__version__) >= version.parse('1.12.0') and version.parse(torch.__version__) < version.parse('2.0.1'):
+            cfg = FullStateDictConfig(offload_to_cpu=True, rank0_only=True)
+            with FSDP.state_dict_type(model, StateDictType.FULL_STATE_DICT, cfg):
+                model_state_dict = model.state_dict()
+        elif version.parse(torch.__version__) >= version.parse('2.0.1'):
+            self.__set_model_optim_state(model, 
+                                         StateDictType.FULL_STATE_DICT,
+                                         FullStateDictConfig(rank0_only=True))
+            model_state_dict = model.state_dict()
+        else:
+            raise RuntimeError("FSDP is not supported while torch version under 1.12.0.")
+        self.save_checkpoint(model_state_dict, checkpoint)
+
+    def save_unsharded_optimizer(self, model: nn.Module, optimizer: Optimizer, checkpoint: str):
+        """
+        Save optimizer to checkpoint but only on master process.
+        """
+
+        if version.parse(torch.__version__) >= version.parse('1.12.0') and version.parse(torch.__version__) < version.parse('2.0.1'):
+            optim_state_dict = FSDP.full_optim_state_dict(model=model, optim=optimizer)
+        elif version.parse(torch.__version__) >= version.parse('2.0.1'):
+            self.__set_model_optim_state(model,
+                                         StateDictType.FULL_STATE_DICT,
+                                         FullOptimStateDictConfig(rank0_only=True))
+            optim_state_dict = FSDP.optim_state_dict(model, optimizer)
+        else:
+            raise RuntimeError("FSDP is not supported while torch version under 1.12.0.")
+        self.save_checkpoint(optim_state_dict, checkpoint)
+
+class TorchFSDPModel(ModelWrapper):
+
+    def __init__(self, module: nn.Module, *args, **kwargs) -> None:
+        super().__init__(module)
+        self.module = FSDP(module, *args, **kwargs)
+
+    def unwrap(self):
+        return self.module.module
+
+
+class TorchFSDPPlugin(DPPluginBase):
+    """
+    Plugin for PyTorch FSDP.
+
+    Example:
+        >>> from colossalai.booster import Booster
+        >>> from colossalai.booster.plugin import TorchFSDPPlugin
+        >>>
+        >>> model, train_dataset, optimizer, criterion = ...
+        >>> plugin = TorchFSDPPlugin()
+
+        >>> train_dataloader = plugin.prepare_train_dataloader(train_dataset, batch_size=8)
+        >>> booster = Booster(plugin=plugin)
+        >>> model, optimizer, train_dataloader, criterion = booster.boost(model, optimizer, train_dataloader, criterion)
+
+    Args:
+        See https://pytorch.org/docs/stable/fsdp.html for details.
+    """
+
+    def __init__(self,
+                 *args,
+                 **kwargs,
```

Comment:
Why do you use `**kwargs`? It's hard for users to configure.

Reply:
The init func of different versions of FSDP have different signatures, the number and types of parameters are different.